plugin/arcane-stream-sqlserver-change-tracking/src/main/scala/models/app/SqlServerChangeTrackingStreamContext.scala

```diff
@@ -0,0 +1,82 @@
+package com.sneaksanddata.arcane.sql_server_change_tracking
+package models.app
+
+import com.sneaksanddata.arcane.framework.models.app.StreamContext
+import com.sneaksanddata.arcane.framework.models.settings.{GroupingSettings, VersionedDataGraphBuilderSettings}
+import com.sneaksanddata.arcane.framework.services.lakehouse.{IcebergCatalogCredential, S3CatalogFileIO}
+import com.sneaksanddata.arcane.framework.services.lakehouse.base.IcebergCatalogSettings
+import com.sneaksanddata.arcane.framework.services.mssql.ConnectionOptions
+import zio.ZLayer
+import zio.json.*
+
+import java.time.Duration
+
+
+/**
+ * The context for the SQL Server Change Tracking stream.
+ * @param spec The stream specification
+ */
+case class SqlServerChangeTrackingStreamContext(spec: StreamSpec) extends StreamContext
+  with GroupingSettings
+  with IcebergCatalogSettings
+  with VersionedDataGraphBuilderSettings:
+
+  implicit val icebergSettingsEncoder: JsonEncoder[CatalogSettings] = DeriveJsonEncoder.gen[CatalogSettings]
+  implicit val specEncoder: JsonEncoder[StreamSpec] = DeriveJsonEncoder.gen[StreamSpec]
+  implicit val contextEncoder: JsonEncoder[SqlServerChangeTrackingStreamContext] = DeriveJsonEncoder.gen[SqlServerChangeTrackingStreamContext]
+
+  override val rowsPerGroup: Int = spec.rowsPerGroup
+  override val lookBackInterval: Duration = Duration.ofSeconds(spec.lookBackInterval)
+  override val changeCaptureInterval: Duration = Duration.ofSeconds(spec.changeCaptureIntervalSeconds)
+  override val groupingInterval: Duration = Duration.ofSeconds(spec.groupingIntervalSeconds)
+
+  override val namespace: String = spec.catalogSettings.namespace
+  override val warehouse: String = spec.catalogSettings.warehouse
+  override val catalogUri: String = spec.catalogSettings.catalogUri
+
+  override val additionalProperties: Map[String, String] = IcebergCatalogCredential.oAuth2Properties
+  override val s3CatalogFileIO: S3CatalogFileIO = S3CatalogFileIO
+
+  override val stagingLocation: Option[String] = spec.stagingLocation
+
+  @jsonExclude
+  val connectionString: String = sys.env("ARCANE_CONNECTIONSTRING")
+
+  val database = "IntegrationTests"
```

Comment:
this looks like it should be provided from spec?

Reply:
This is correct